src/lib/FileDownload.ts

```diff
@@ -125,12 +181,24 @@ class FileDownload extends EventEmitter
             });
 
             // Everything else happens after we get a response -----------------
-            downloadRequest.on("response", res => {
-                
+            downloadRequest.on("response", (res: Response) => {
+                const delay = this.calculateRetryDelay({
+                    attemptCount: this.state.numTries,
+                    retryOptions: res.request.options.retry,
+                    response: res
+                })
+                if (delay > 0) {
+                    return resolve(wait(delay, signal).then(() => {
+                        // Destroy this current request before making another one
+                        downloadRequest.destroy()
+                        return this.run(options)
+                    }))
```

Comment:
Calling `resolve` with a promise that might fail feels odd. What about
```
return wait(delay, signal).then(() => {
    downloadRequest.destroy()
    return this.run(options)
}).then(resolve, reject)
```

Reply:
Great point!